crypto/bn/bn_lib.c

```diff
@@ -903,6 +903,34 @@ void BN_consttime_swap(BN_ULONG condition, BIGNUM *a, BIGNUM *b, int nwords)
     a->top ^= t;
     b->top ^= t;
 
+    t = (a->neg ^ b->neg) & condition;
+    a->neg ^= t;
+    b->neg ^= t;
+
+    /*-
+     * Idea behind BN_FLG_STATIC_DATA is actually to
+     * indicate that data may not be written to.
+     * Intention is actually to treat it as it's
+     * read-only data, and some (if not most) of it does
+     * reside in read-only segment. In other words
+     * observation of BN_FLG_STATIC_DATA in
+     * BN_consttime_swap should be treated as fatal
+     * condition. It would either cause SEGV or
+     * effectively cause data corruption.
+     * BN_FLG_MALLOCED refers to BN structure itself,
+     * and hence must be preserved. Remaining flags are
+     * BN_FLG_CONSTIME and BN_FLG_SECURE. Latter must be
+     * preserved, because it determines how x->d was
+     * allocated and hence how to free it. This leaves
+     * BN_FLG_CONSTTIME that one can do something about.
+     * To summarize it's sufficient to mask and swap
+     * BN_FLG_CONSTTIME alone. BN_FLG_STATIC_DATA should
+     * be treated as fatal.
+     */
+    t = ((a->flags ^ b->flags) & BN_FLG_CONSTTIME) & condition;
+    a->flags ^= t;
+    b->flags ^= t;
+
```

Comment:
`BN_FLG_FIXED_TOP` indicates that we haven't called `bn_correct_top()` on the data, so the d array may be padded with additional 0 values (i.e. top could be greater than the minimal value that it could be). So I think we should be swapping it. Does that suggest we need to make an equivalent change in master/1.1.1?

Reply:
> Does that suggest we need to make an equivalent change in master/1.1.1?

I reckon so. Seems like it came from 327b2c011342280c7fd5e312a4fff2a01083d2d6 -- @dot-asm any comment?

Reply:
Yes, it's the right thing to do. Just in case for reference, BN_FLG_FIXED_TOP is a very special *debugging* flag and is customarily zero. So that production builds won't be affected. It's *not* suggestion to not resolve this, only explanation to it can go unnoticed.

Reply:
Thanks @dot-asm ! I'll open a PR for master.

Reply:
While we are at it, and while you'll be at it, I'd like to suggest to a) reformat commentary prior BN_consttime_swap to fit 80 chars; b) omit switch at the end and replace it with straight loop. Rationale for b) is that compilers are in better position to decide how to handle loops, nor are straight loops hopeless on contemporary processors. I mean straight loop in machine code is not unlikely to perform better than the current loop-switch combo.